batteries/colorful.luau

```diff
@@ -1,27 +1,69 @@
+local process = (require)("@lute/process") -- temporary ()
+
+export type Styler = (text: string) -> string
+
+-- https://no-color.org/
+local function colorDisabled(): boolean
+	-- is it really okay to index env every style call?
+	return process.env.NO_COLOR ~= nil
+end
+
+local function searchForSubstringIn(str: string, substring: string, index: number)
+	-- last argument of string.find() turns string.find into plain substring search
+	return string.find(str, substring, index, true)
+end
+
 local function replaceClose(str: string, close: string, replace: string, index: number?): string
 	local result = ""
 	local cursor = 1
+
 	while index do
 		result ..= string.sub(str, cursor, index - 1) .. replace
+
 		cursor = index + #close
-		local nextIndex = string.find(str, close, cursor, true)
-		index = nextIndex
+
+		index = searchForSubstringIn(str, close, cursor)
 	end
+
 	return result .. string.sub(str, cursor)
 end
 
-local function formatter(open: string, close: string, replace: string?): (string) -> string
-	return function(str: string): string
-		local index = string.find(str, close, #open + 1, true)
-		if index then
-			return open .. replaceClose(str, close, replace or open, index) .. close
-		else
-			return open .. str .. close
+local function formatter(open: string, close: string, replace: string?): Styler
+	local replaceCloseWith = replace or open
+
+	local function created_formatter(str: string): string
+		if colorDisabled() then
+			return str
 		end
+
+		local index = searchForSubstringIn(str, close, #open + 1)
+		local encasedString = if index then replaceClose(str, close, replaceCloseWith, index) else str
+
+		return `{open}{encasedString}{close}`
```

Comment:
just a discussion point, why do we prefer string interpolation here? I don't personally find it easier to read and only ever use it when mixing with inline strings, but this doesn't have that.

Reply:
To be honest, I've always just kind of had this idea that it was better but never really thought about why. I don't think it matters. Personally I find it more readable but that's probably because of my own consistency

Reply:
I think I prefer just concat for things that are literally just concat.